File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        if len(numbers) < 2:
            return numbers
        
        left = 0
        right = len(numbers) - 1

        while left < right:
            sum = numbers[left] + numbers[right]

            if (sum == target):
                return [left + 1, right + 1]
            
            elif sum > target: 
                right -= 1
            else:
                left += 1
        

File: Data Structures & Algorithms/max-water-container/submission-2.ts
class Solution {
    /**
     * @param {number[]} heights
     * @return {number}
     */
    maxArea(heights: number[]): number {
        let left = 0;
        let right = heights.length - 1;

        let result = 0;

        while (left < right) {
            let lowHeight = Math.min(heights[left], heights[right]);

            let distance = right - left;

            let area = lowHeight * distance;

            console.log(area);

            result = Math.max(area, result);

            if (heights[left] > heights[right]) {
                right--;
            } else {
                left++;
            }
        }

        return result;
    }
}
